lib/__init__.py

```diff
@@ -1,11 +1,13 @@
+from __future__ import absolute_import
+
 import os
 import sys
 import shutil
 
-from lib.options import Options
-from lib.tarantool_server import TarantoolServer
-from lib.unittest_server import UnittestServer
-from utils import warn_unix_sockets_at_start
+from .options import Options
+from .tarantool_server import TarantoolServer
+from .unittest_server import UnittestServer
+from .utils import warn_unix_sockets_at_start
```

Comment:
This line is actually becomes broken on Python 3 and there are two ways to fix. Let's looks at cons and pros of both approaches.

## from lib.utils import warn_unix_sockets_at_start

Pros:

1. Consistent with existing code.
2. Minimal change (only broken lines will be changed).
3. It follows majority of the existing code (we already use `from lib.foo import bar` in `lib/*.py` in the vast majority of imports).
4. Easy to explain.

Cons:

1. Using relative import would be nice, it looks a bit more natural.

(The patch below may require small tweaks depending on a place in git history, where we intend to apply it.)

<details>
<summary>Patch</summary>

```diff
commit d45edc0b22dfb2280ac7051b102bfc85a9861a49
Author: Alexander Turenko <alexander.turenko@tarantool.org>
Date:   Fri Feb 19 03:32:26 2021 +0300

    python3: get rid of package level imports
    
    Unlike Python 2, Python 3 does not treat `import bar` from a module in a
    package `foo` as importing of `foo.bar`.
    
    Absolute import that is relative to a test-run root (like `import
    lib.utils`) still work, because `sys.path` contains a script directory
    (after symlinks resolution) on both Python versions.
    
    https://www.python.org/dev/peps/pep-0328/

diff --git a/lib/__init__.py b/lib/__init__.py
index c03a6d8..0dffc40 100644
--- a/lib/__init__.py
+++ b/lib/__init__.py
@@ -5,7 +5,7 @@ import shutil
 from lib.options import Options
 from lib.tarantool_server import TarantoolServer
 from lib.unittest_server import UnittestServer
-from utils import warn_unix_sockets_at_start
+from lib.utils import warn_unix_sockets_at_start
 
 
 __all__ = ['Options']
diff --git a/lib/admin_connection.py b/lib/admin_connection.py
index 062d7c7..5551437 100644
--- a/lib/admin_connection.py
+++ b/lib/admin_connection.py
@@ -24,9 +24,9 @@ __author__ = "Konstantin Osipov <kostja.osipov@gmail.com>"
 import re
 import sys
 
-from tarantool_connection import TarantoolConnection
-from tarantool_connection import TarantoolPool
-from tarantool_connection import TarantoolAsyncConnection
+from lib.tarantool_connection import TarantoolConnection
+from lib.tarantool_connection import TarantoolPool
+from lib.tarantool_connection import TarantoolAsyncConnection
 
 if sys.version[0] == '2':
     reload(sys)         # noqa: F821
diff --git a/lib/app_server.py b/lib/app_server.py
index a37ef18..2343a0b 100644
--- a/lib/app_server.py
+++ b/lib/app_server.py
@@ -21,7 +21,7 @@ from lib.utils import find_port
 from lib.utils import format_process
 from lib.utils import signame
 from lib.utils import warn_unix_socket
-from test import TestRunGreenlet, TestExecutionError
+from lib.test import TestRunGreenlet, TestExecutionError
 from threading import Timer
 
 if sys.version[0] == '2':
diff --git a/lib/box_connection.py b/lib/box_connection.py
index dad84a2..9dd212d 100644
--- a/lib/box_connection.py
+++ b/lib/box_connection.py
@@ -25,7 +25,7 @@ import errno
 import ctypes
 import socket
 
-from tarantool_connection import TarantoolConnection
+from lib.tarantool_connection import TarantoolConnection
 
 # monkey patch tarantool and msgpack
 from lib.utils import check_libs
diff --git a/lib/connpool.py b/lib/connpool.py
index 3e9d654..b80ba58 100644
--- a/lib/connpool.py
+++ b/lib/connpool.py
@@ -10,7 +10,7 @@ from collections import deque
 from contextlib import contextmanager
 from functools import wraps
 
-from test import TestRunGreenlet
+from lib.test import TestRunGreenlet
 
 __all__ = ["ConnectionPool", "retry"]
 
diff --git a/lib/tarantool_connection.py b/lib/tarantool_connection.py
index f2a5267..b246d4f 100644
--- a/lib/tarantool_connection.py
+++ b/lib/tarantool_connection.py
@@ -30,10 +30,10 @@ from contextlib import contextmanager
 import gevent
 from gevent import socket as gsocket
 
-from connpool import ConnectionPool
-from test import TestRunGreenlet
-from utils import warn_unix_socket
-from utils import set_fd_cloexec
+from lib.connpool import ConnectionPool
+from lib.test import TestRunGreenlet
+from lib.utils import warn_unix_socket
+from lib.utils import set_fd_cloexec
 
 
 class TarantoolPool(ConnectionPool):
diff --git a/lib/tarantool_server.py b/lib/tarantool_server.py
index 47631df..38a1ab3 100644
--- a/lib/tarantool_server.py
+++ b/lib/tarantool_server.py
@@ -35,7 +35,7 @@ from lib.utils import safe_makedirs
 from lib.utils import signame
 from lib.utils import warn_unix_socket
 from lib.utils import prefix_each_line
-from test import TestRunGreenlet, TestExecutionError
+from lib.test import TestRunGreenlet, TestExecutionError
 
 try:
     # Python 2
diff --git a/lib/test.py b/lib/test.py
index e6192b5..0fc4c02 100644
--- a/lib/test.py
+++ b/lib/test.py
@@ -2,20 +2,20 @@ import filecmp
 import gevent
 import os
 import pprint
-import pytap13
 import re
 import shutil
 import sys
 import traceback
 from functools import partial
 from hashlib import md5
-from utils import safe_makedirs
 
+from lib import pytap13
 from lib import Options
 from lib.colorer import color_stdout
 from lib.utils import non_empty_valgrind_logs
 from lib.utils import print_tail_n
 from lib.utils import print_unidiff as utils_print_unidiff
+from lib.utils import safe_makedirs
 
 if sys.version[0] == '2':
     reload(sys)     # noqa: F821
```
</details>

## from .utils import warn_unix_sockets_at_start

Pros:

1. This way looks a bit more sane and natural.

Cons:

1. All other imports should be changed in the same way for consistency.
2. So the patch will change not only problematic imports, but all, and it'll be harder to spot where a problem actually was.
3. And so we should cleanly describe the problem, alternative solution and a reason why we decide to use relative imports in the commit message.

## My thoughts

Personally I would prefer the minimal change. However I'll accept any way if it is done consistently and the commit message responds to questions 'what was broken' and 'why it is fixed this way'.